ui/raidboss/emulator/data/LogEventHandler.ts

```diff
@@ -0,0 +1,119 @@
+import EmulatorCommon from '../EmulatorCommon';
+import EventBus from '../EventBus';
+import { LocaleNetRegex } from '../../../../resources/translations';
+import { Lang } from 'types/global';
+import LineEvent from './network_log_converter/LineEvent';
+import { LineEvent0x01 } from './network_log_converter/LineEvent0x01';
+
+export default class LogEventHandler extends EventBus {
+  static doesLineMatch(line: string, regexes: Record<string, RegExp>): RegExpExecArray | false {
```

Comment:
Can this be `RegExpExecArray | undefined` here?

Also, can it be `regexes: Record<Lang, RegExp> | RegExp`? It seems like it'd be better for the callers to do something like `res = LogEventHandler.doesLineMatch(line, EmulatorCommon.cactbotWipeRegex);` rather than the `{ '': regex }` construction.  All of this would also let you remove the `as Lang` and the `regexes[i]?` below.

Reply:
Switching to `undefined` makes sense here, yeah. I don't know how to make `regexes: Record<Lang, RegExp> | RegExp` work in the scope of the function though.

Reply:
```typescript
  static doesLineMatch(line: string,
      regexes: Record<Lang, RegExp> | RegExp): RegExpExecArray | null {
    if (regexes instanceof RegExp)
      return regexes.exec(line);

    for (const langStr in regexes) {
      const lang = langStr as keyof typeof regexes;
      const res = regexes[lang].exec(line);
      if (res) {
        if (LocaleNetRegex.areaSeal[lang] && res.groups)
          res.groups.language = lang;
        return res;
      }
    }
    return null;
  }
```

Reply:
Actually `LocaleNetRegex.areaSeal[lang]` is always true, so that can be elided too.

Reply:
Thanks for this. I don't know why, but I was absolutely failing to string this one together. Adjusted for next commit, including the `areaSeal` removal.